core/src/main/java/com/datastax/oss/driver/api/core/metadata/diagnostic/TopologyDiagnostic.java

```diff
@@ -0,0 +1,135 @@
+  NodeGroupDiagnostic getGlobalDiagnostic();
+
+  /**
+   * Returns datacenter-specific {@link NodeGroupDiagnostic} instances, keyed by datacenter name.
+   */
+  @NonNull
+  Map<String, NodeGroupDiagnostic> getLocalDiagnostics();
+
+  /**
+   * Returns the status of the cluster topology. By default this is the same as {@code
+   * getGlobalDiagnostic().getStatus()}. See {@link NodeGroupDiagnostic#getStatus()} for details
+   * about possible statuses and their meanings.
+   */
+  @NonNull
+  @Override
+  default Status getStatus() {
+    return getGlobalDiagnostic().getStatus();
+  }
+
+  @NonNull
+  @Override
+  default Map<String, Object> getDetails() {
+    return ImmutableMap.<String, Object>builder()
+        .put("status", getStatus())
+        .putAll(getGlobalDiagnostic().getDetails())
+        .putAll(
+            getLocalDiagnostics().entrySet().stream()
+                .collect(
+                    ImmutableMap.toImmutableMap(
+                        Entry::getKey, entry -> entry.getValue().getDetails())))
+        .build();
+  }
+
+  /**
+   * A health {@link Diagnostic} for a group of nodes, detailing how many nodes were found in total,
+   * how many were up, and how many were down.
+   *
+   * <p>The breadth of the report's scope depends on how it was created; it may refer to the entire
+   * cluster, or to just a datacenter.
+   */
+  interface NodeGroupDiagnostic extends Diagnostic {
+
+    /** @return the total number of nodes in this group. */
+    int getTotal();
+
+    /**
+     * Returns the number of nodes in this group whose {@linkplain
+     * com.datastax.oss.driver.api.core.metadata.Node#getState() state} is {@link
+     * com.datastax.oss.driver.api.core.metadata.NodeState#UP UP}.
+     *
+     * @return the number of nodes in this group that are known to be up.
+     */
+    int getUp();
+
+    /**
+     * Returns the number of nodes in this group whose {@linkplain
+     * com.datastax.oss.driver.api.core.metadata.Node#getState() state} is {@link
+     * com.datastax.oss.driver.api.core.metadata.NodeState#DOWN DOWN} or {@link
+     * com.datastax.oss.driver.api.core.metadata.NodeState#FORCED_DOWN FORCED_DOWN}.
+     *
+     * @return the number of nodes in this group that are known to be down.
+     */
+    int getDown();
+
+    /**
+     * Returns the number of nodes in this group whose {@linkplain
+     * com.datastax.oss.driver.api.core.metadata.Node#getState() state} is {@link
+     * com.datastax.oss.driver.api.core.metadata.NodeState#UNKNOWN UNKNOWN}.
+     *
+     * <p>Nodes may be in an unknown state if the driver hasn't connected to them at all, and it has
+     * not received any Gossip event indicating their actual state. Most of such nodes should
+     * actually be up.
+     *
+     * @return the number of nodes in this group that are in unknown state.
+     */
+    int getUnknown();
+
+    /**
+     * Returns the status of this node group. The status will be {@link Status#AVAILABLE} if all
```

Comment:
After changing topology diagnostics to adopt a pessimistic approach to unknown nodes, I think we should do the same here. I will work on it today.